src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -118,6 +118,9 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
 {
     LogQueryImageResponse(response);
 
+    uint32_t currentSoftwareVersion;
+    VerifyOrReturn(Basic::Attributes::SoftwareVersion::Get(kRootEndpointId, &currentSoftwareVersion) == EMBER_ZCL_STATUS_SUCCESS);
```

Comment:
We shouldn't actually return on error. We must report all errors to the OTA Requestor Driver so that it knows that the current update failed and it can schedule another one. Maybe even `VerifyOrDie` would be a good choice here as the error clearly means that the app has a wrong ZAP config.